datadog/tests/resource_datadog_role_test.go

```diff
@@ -171,16 +222,26 @@ resource "datadog_role" "foo" {
 }`, uniq)
 }
 
-func testAccCheckDatadogRoleConfigNoPerm(uniq string) string {
+func testAccCheckDatadogRoleConfigNoUnrestrictedPerm(uniq string) string {
+	return fmt.Sprintf(`
+data "datadog_permissions" foo {}
+
+resource "datadog_role" "foo" {
+  name      = "%s"
+}`, uniq)
+}
+
+func testAccCheckDatadogRoleConfigNoRestrictedPerm(uniq string) string {
 	return fmt.Sprintf(`
 data "datadog_permissions" foo {}
 
 resource "datadog_role" "foo" {
   name      = "%s"
+  default_permissions_opt_out = true
 }`, uniq)
 }
 
-func testAccCheckDatadogRoleConfigRestrictedPerm(uniq string) string {
```

Comment:
hmm this test name is curious. it wasn't configuring with a restricted permission but a nonexistent permission. tho i guess if the permission data source didn't include restricted permissions they would've been "invalid" permission names

Reply:
yeah lol, i was also initially confused by the test name (hence the rename + addition of a new test)